.pre-commit-config.yaml

```diff
@@ -7,23 +8,64 @@ repos:
     types_or: [file]
     files: \.(cpp|def|h|cu|tpp|kernel|loader|unitless|hpp|param)$
 - repo: https://github.com/pre-commit/pre-commit-hooks
-  rev: v2.3.0
+  rev: v4.5.0
   hooks:
   - id: no-commit-to-branch
     args: [-b, dev]
   - id: check-merge-conflict
   - id: trailing-whitespace
     exclude_types: [markdown, rst]
   - id: end-of-file-fixer
-- repo: https://github.com/PyCQA/flake8
-  rev: 7.0.0
+  - id: check-toml
+  - id: check-yaml
+  - id: mixed-line-ending
+  - id: check-executables-have-shebangs
+  - id: check-shebang-scripts-are-executable
+    exclude: \.tpl$
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  rev: v1.5.4
   hooks:
-  - id: flake8
-    # black's recommendation on compatibility with flake8
-    # see https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#flake8
-    args: [--extend-ignore, E203, --max-line-length, "120"]
-- repo: https://github.com/psf/black-pre-commit-mirror
-  rev: 23.12.1
+    - id: forbid-tabs
+      types_or: [file]
+      exclude_types: [rst, tex, tsv]
+    - id: remove-tabs
+      types_or: [file]
+      exclude_types: [rst, tex, tsv]
+    - id: forbid-crlf
+    - id: remove-crlf
+- repo: https://github.com/brutus/enforce-ascii
+  rev: v0.2.1
   hooks:
-  - id: black
-    args: [-l, "120"]
+      - id: enforce-ascii
+        types_or: [file]
+        exclude_types: [rst, tex, tsv, jupyter, markdown]
+        exclude: .zenodo.json
+- repo: local
+  hooks:
+    - id: check_cpp_code_style
+      name: check_cpp_code_style
+      entry: share/ci/check_cpp_code_style.sh
+      language: system
+      files: \.(cpp|def|h|cu|tpp|kernel|loader|unitless|hpp|param)$
+      pass_filenames: false
+      # This hook runs on all files which is why it feels a bit too heavy-weight
+      # for running before every commit. The CI currently runs manually, so it's
+      # still running there and can be run manually on demand.
+      # TODO: Adjust the script to take filenames, so that we can pass the
+      # filenames and speed things up before commits.
+      stages: [manual]
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.2.1
+  hooks:
+    # Run the linter.
+    - id: ruff
+      types_or: [ python, pyi, jupyter ]
+      # Rule E721 is about using isinstance()
+      # but lib/python/picongpu/pypicongpu/rendering/renderer.py
+      # is strict about types. Could be solved locally but I think this is fine.
+      args: [ --fix, --ignore, E721]
+    # Run the formatter.
+    - id: ruff-format
+      types_or: [ python, pyi, jupyter ]
+      args: ["--line-length", "120"]
```

Comment:
This is a change from our previous 79 characters per line (PEP8) limit

Reply:
Yes, this was agreed with @psychocoderHPC and I thought that it was already agreed upon elsewhere. But I've just tried to look it up but admittedly I could neither find it in `CONTRIBUTING.md` nor in our [Contributing repo](https://github.com/ComputationalRadiationPhysics/contributing). No strong feelings here. Actually, I would personally not change it.